src/components/elements/square-icon/square-icon.styles.ts

```diff
@@ -0,0 +1,69 @@
+'use client';
+
+import type { SquareIconProps } from './square-icon.types';
+
+import { css, styled } from '@/styles';
+import { adjustOpacity } from '@/styles/helpers/adjust-opacity';
+
+export const StyledSquareIcon = styled.div<Partial<SquareIconProps>>`
+  display: flex;
+  width: fit-content;
+  border-radius: 12px;
+  padding: 12px;
+  border: 1px solid ${({ theme }) => adjustOpacity(theme.colors.white, 0.14)};
+
+  ${({ variant, size }) => {
+    switch (variant) {
+      case 'primary':
+        return css`
+          background-color: ${({ theme }) => theme.colors.inkBlack};
+
+          & svg {
+            color: ${({ theme }) => theme.colors.electricBlue};
+            box-shadow: 0px 2.4000000953674316px 1.2000000476837158px 0px #c9d4fb1a inset;
+            box-shadow: 0px -1.2000000476837158px 1.2000000476837158px 0px #102e971a inset;
+            box-shadow: 0px 2.4000000953674316px 1.2000000476837158px 0px #c9d4fb1a inset;
+            box-shadow: 0px -1.2000000476837158px 1.2000000476837158px 0px #102e971a inset;
+          }
+
+          ${() =>
+            size === 'large' &&
```

Comment:
Does Figma count the border? because by the measures I don't think could be possible 56 with all the measures. content 32px + 24px (12 each side) padding + 2px (1 each side) border = 58px

Reply:
With the border should be 56px. The border that has in Figma is inside, not outside, so I think you can do it with shadow instead of border to avoid the extra space.

<img width="698" alt="Screenshot 2024-02-23 at 13 37 52" src="https://github.com/bejamas/httptoolkit-website/assets/22168189/9c04bb41-9d1e-46b9-ad67-dd3fdc8c05db">

Reply:
Got ti